File: algorithms/14.py
# https://leetcode.com/problems/longest-common-prefix


class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if strs == []:
            return ""

        minStrLength = min(map(len, strs))
        for i in range(minStrLength):
            if len(set(map(lambda s: s[i], strs))) != 1:
                return strs[0][:i]
        return strs[0][:minStrLength]

File: algorithms/14.swift
// https://leetcode.com/problems/longest-common-prefix

class Solution {
    func longestCommonPrefix(strs: [String]) -> String {
        guard strs.count > 0 else { return "" }

        let minStrLength = strs.map { $0.characters.count }.reduce(Int.max, combine: min)
        for i in 0..<minStrLength {
            if Set(strs.map { $0.characters[$0.characters.startIndex.advancedBy(i)] }).count > 1 {
                return strs[0].substringToIndex(strs[0].characters.startIndex.advancedBy(i))
            }
        }
        return strs[0].substringToIndex(strs[0].characters.startIndex.advancedBy(minStrLength))
    }
}
